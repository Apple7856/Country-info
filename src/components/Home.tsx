import { Button, Container, makeStyles, TextField, Typography } from '@material-ui/core';
import axios from 'axios';
import React, { useEffect, useState } from 'react';

const useStyle = makeStyles({
    home: {
        height: '100vh',
        display: 'flex',
        flexDirection: 'column',
        alignItems: "center",
    },
    form: {
        marginTop: "20px"
    },
    button: {
        marginTop: "20px"
    },
    capitalContainer: {
        width: "600px",
        height: "250px",
        borderRadius: "5px",
        display: "flex",
        flexDirection: "column",
        alignItems: "center",

    }
})

interface state {
    open: boolean,
    submitOpen: boolean,
    capitalWeather: boolean,
    inputVal: any,
    cityData: any,
    locationData: any,
}

const Home: React.FC = () => {
    const classes = useStyle();
    const [open, setOpen] = useState(false);
    const [capitalWeather, setCapitalWeather] = useState(false);
    const [inputVal, setInputVal] = useState("");
    const [cityData, setCityData] = useState("");
    const [locationData, setLocationData] = useState("");

    const submitClick = async () => {
        try {
            const data = await axios.get(`http://api.weatherstack.com/current?access_key=8103d12609124071f6e54b72cb853d87&query=${inputVal}`);
            setCityData(data.data.location);
            setLocationData(data.data.current)
            setOpen(true);
            setInputVal("");
            setCapitalWeather(false);
        } catch (error) {
            console.log("Wrong data",error);
        }
    }

    return (
        <Container>
            <Typography component="div" className={classes.home}>
                <form className={classes.form} noValidate autoComplete="off">
                    <TextField id="outlined-basic" value={inputVal} placeholder="Search Country Name" variant="outlined" onChange={(e) => setInputVal(e.target.value)} />
                </form>
                <Button className={classes.button} variant="contained" color="primary" disabled={inputVal ? false : true} onClick={() => submitClick()}>
                    Submit
                </Button>
                {
                    open ?
                        <Typography component="div" className={classes.capitalContainer}>
                            <Typography className={classes.button} component="h4">Capital: {cityData.name}</Typography>
                            <Typography className={classes.button} component="h4">Population: { cityData.lon }</Typography>
                            <Typography className={classes.button} component="h4">latlng: { cityData.lat }</Typography>
                            <Typography className={classes.button} component="h4">Country: {cityData.country} </Typography>
                            <Button className={classes.button} variant="contained" color="primary" onClick={() => setCapitalWeather(!capitalWeather)}>
                                Capital Weather
                            </Button>
                        </Typography>
                        : ""
                }
                {
                    capitalWeather ?
                        <Typography component="div" className={classes.capitalContainer}>
                            <Typography className={classes.button} component="h4">Temperature: { locationData.temperature }</Typography>
                            <Typography className={classes.button} component="h4">Weather Icons: <img src={locationData.weather_icons[0]} /></Typography>
                            <Typography className={classes.button} component="h4">Wind Speed: { locationData.wind_speed }</Typography>
                            <Typography className={classes.button} component="h4">Precip: { locationData.precip}</Typography>
                        </Typography>
                        : ""
                }
            </Typography>
        </Container>
    )
}

export default Home
