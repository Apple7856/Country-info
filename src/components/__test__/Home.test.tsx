import Home from "../Home";
import { act, cleanup, fireEvent, render } from '@testing-library/react';
import renderer from 'react-test-renderer';

describe("Home Component", () => {

    afterEach(() => {
        cleanup();
    })

    test("Home Component", async () => {
        const { getByTestId } = render(<Home />);
        const mainComponent = getByTestId("mainComponent");
        expect(mainComponent).toBeTruthy();
    })

    test("Input TextField", async () => {
        await act(async () => {
            const { getByTestId } = render(<Home />);
            const inputField = getByTestId("inputField");
            expect(inputField).toBeTruthy();
        })
    })

    test("Button Test", () => {
        const home = render(<Home />);
        const buttonTest = home.getByTestId("buttonTest");
        expect(buttonTest).toBeTruthy();
    })

    test("Matches SnapShot", () => {
        const tree = renderer.create(<Home />).toJSON();
        expect(tree).toMatchSnapshot();
    })

})
