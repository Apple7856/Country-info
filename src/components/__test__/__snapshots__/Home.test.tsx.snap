// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component Matches SnapShot 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg"
>
  <div
    className="MuiTypography-root makeStyles-home-25 MuiTypography-body1"
    data-testid="mainComponent"
  >
    <form
      autoComplete="off"
      className="makeStyles-form-26"
      noValidate={true}
    >
      <div
        className="MuiFormControl-root MuiTextField-root"
        data-testid="inputField"
      >
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            id="outlined-basic"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Search Country Name"
            required={false}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-29 MuiOutlinedInput-notchedOutline"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="PrivateNotchedOutline-legend-30"
              style={
                Object {
                  "width": 0.01,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
    </form>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-27 MuiButton-containedPrimary Mui-disabled Mui-disabled"
      data-testid="buttonTest"
      disabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={-1}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Submit
      </span>
    </button>
    
  </div>
</div>
`;
